caravel/assets/javascripts/dashboard.jsx

```diff
@@ -182,15 +167,148 @@ class GridLayout extends React.Component {
         margin={[20, 20]}
         useCSSTransforms={false}
         draggableHandle=".drag">
-        {this.state.sliceElements}
+        {this.state.slices.map(function (slice) {
+          return (
+            <div
+              id={"slice_" + slice.slice_id}
+              key={slice.slice_id}
+              data-slice-id={slice.slice_id}
+              className={"widget " + slice.viz_name}>
+              <SliceCell
+                slice={slice}
+                removeSlice={this.removeSlice.bind(this)}
+                expandedSlices={this.props.dashboard.metadata.expanded_slices}/>
+            </div>
+          );
+        }, this)}
       </ResponsiveReactGridLayout>
     );
   }
 }
 
-var Dashboard = function (dashboardData) {
-  var reactGridLayout;
+class SliceAdder extends React.Component {
```

Comment:
we should pull this component out into it's own file then import it.

maybe in a new folder at `caravel/assets/javascripts/dashboard/components`.